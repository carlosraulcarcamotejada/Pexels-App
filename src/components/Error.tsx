import { ExclamationCircleIcon } from '@heroicons/react/solid';
import { motion } from 'framer-motion';
import {FC} from 'react'

type props = {
    message:string
}

const Error:FC<props> = ({ message }):JSX.Element => {
  return (
    <motion.div
      initial={{opacity:0}}
      animate={{opacity:1}}
      transition={{duration:0.4}}
      className="w-90 h-90 bg-red-300/30 flex justify-center text-sm font-semibold text-red-500">
        <ExclamationCircleIcon className="w-5 h-5 mr-2" />
        {message}
    </motion.div>
  );
}

export default Error