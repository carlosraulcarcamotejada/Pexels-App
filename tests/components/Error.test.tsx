import { render } from "@testing-library/react"
import Error from "../../src/components/Error"
import React from "react"

describe('Pruebas en <Error />', () => { 
    test('debe de hacer match con el snapshot', () => { 

        const message = 'This is an error'
        const {container} = render(<Error message={message} />)
        expect(container).toMatchSnapshot()
     })
 })