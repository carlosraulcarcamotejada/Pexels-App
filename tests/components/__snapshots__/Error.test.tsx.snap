// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Error /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="w-90 h-90 bg-red-300/30 flex justify-center text-sm font-semibold text-red-500"
    style="opacity: 0;"
  >
    <svg
      aria-hidden="true"
      class="w-5 h-5 mr-2"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    This is an error
  </div>
</div>
`;
